coda_v2_python_client/firebase_client_wrapper.py

```diff
@@ -756,35 +756,44 @@ def add_message_to_dataset(self, dataset_id, message, max_segment_size=MAX_SEGME
         :type dataset_id: str
         :param message: The message to be added.
         :type message: core_data_modules.data_models.message.Message
-        :param max_segment_size: the maximum size for a segment, defaults to 2500
+        :param max_segment_size: The maximum size for a segment, defaults to 2500
         :type max_segment_size: int, optional
         """
-        message_exists = self.get_dataset_message(dataset_id, message.message_id) is not None
-        assert not message_exists, f"message with id {message.message_id} already exists."
+        @firestore.transactional
+        def add_in_transaction(transaction, message=message):
+            message_id = message.message_id
```

Comment:
Can you insert a log along the lines of "Adding message with id {} to Coda dataset {}"

Reply:
Updated, thanks.